### PR TITLE
Fixing a bug in JuiceStream hyper generating

### LINKED ISSUE #2586: Hyper dash generation is incorrect on JuiceStreams
When I'm trying to do implements on catch difficulty calculation, I noticed that there's a serious issue on hyper calculating, since in lazer codes droplets are changed to **JuiceStreams**, the hypers' are not being correctly calculated in **CatchBeatmapProcessor** since the processor is from old codes.

Listing the issue here and I'm trying to fix it.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/58565253-hyper-dash-generation-is-incorrect-on-juicestreams?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by frankhjwx**
Resolves #2586.

### CODE REVIEW

**Comment by Aergwyn** [comment]
You can mark issues to be closed with this PR when using certain keywords such as `closes`, `fixes`, `solves` etc. See [here for doc](https://help.github.com/articles/closing-issues-using-keywords/).
Good luck. :3c

**Comment by frankhjwx** [comment]
@Aergwyn thx! I'll use these key words later~

File: osu.Game.Rulesets.Catch/Beatmaps/CatchBeatmapProcessor.cs [hunk 1]
```diff
@@ -33,41 +33,58 @@ public override void PostProcess()
         private void initialiseHyperDash(List<CatchHitObject> objects)
         {
             // todo: add difficulty adjust.
-            double halfCatcherWidth = CatcherArea.CATCHER_SIZE * (objects.FirstOrDefault()?.Scale ?? 1) / CatchPlayfield.BASE_WIDTH / 2;
+            double catcherWidth = (1.0f - 0.7f * (Beatmap.BeatmapInfo.BaseDifficulty.CircleSize - 5) / 5) * 0.62064f;
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/Beatmaps/CatchBeatmapProcessor.cs]: what is `0.62064f`?

**Comment by Hanamuke** [comment]
osu!catch is unplayable as is, and I see that the original author hasn't been active lately. Is it OK if I try to rewrite/cleanup this PR ?  If someone could enlighten me on why a -4 is necessary on this line that would be extra helpful thx : 
` double timeToNext = nextObject.StartTime - currentObject.StartTime - 4;`

**Comment by peppy** [comment]
Yep, go for it. I can provide sample code from stable that may be helpful.

**Comment by Hanamuke** [comment]
Sure why not. I find it very wierd that hyperdash placement should depend on CS tbh, i'm top 1000 ctb and i've never seen hyperjumps placed differently whith HR. I intend to look at ctb playfield width too. The problems maybe related, if you have code on that too.

**Comment by peppy** [comment]
I've already done directly pixel comparisons for what it's worth. I believe if anything, the scale may be slightly off (as well as the scroll speed), but the "width" should be correct.

**Comment by Xinnoh** [comment]
hyperdash placement significantly depends on circle size ._.
HR is not a good example to compare with because fruits move left/right randomely, which is a larger factor than circle size. 
However, pattern changes stand out significantly when playing with EZ. Because fruits are larger, hyperdashes near the trigger distance become edge dashes, the distance between the edges of the fruits is smaller and catchable with normal dashing.

**Comment by peppy** [comment]
It his PR no longer relevant?

**Comment by Hanamuke** [comment]
Yes you can close this PR

**Pull request closed without merging**